### DISCUSSION TITLE
Getting error while running npm i && npm link

### DISCUSSION THREAD

**Discussion by vedantjk** [saran-surya/email_auth_node#34] [Q&A] [answered]
So first error was that it did not recognize '&&' so I used this - '(npm i) -and (npm link)'
PS C:\Users\Vedant\Downloads\ServerForOTP-main> (npm i) -and (npm link)
npm WARN saveError ENOENT: no such file or directory, open 'C:\Users\Vedant\Downloads\ServerForOTP-main\package.json'
npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\Vedant\Downloads\ServerForOTP-main\package.json'
npm WARN ServerForOTP-main No description
npm WARN ServerForOTP-main No repository field.
npm WARN ServerForOTP-main No README data
npm WARN ServerForOTP-main No license field.

npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path C:\Users\Vedant\Downloads\ServerForOTP-main\package.json
npm ERR! errno -4058
npm ERR! enoent ENOENT: no such file or directory, open 'C:\Users\Vedant\Downloads\ServerForOTP-main\package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Vedant\AppData\Roaming\npm-cache\_logs\2022-06-13T13_26_33_425Z-debug.log
False
[2022-06-13T13_30_39_022Z-debug.log](https://github.com/saran-surya/email_auth_node/files/8891485/2022-06-13T13_30_39_022Z-debug.log)
I have attached the debug log as well. I'm new to Flutter and only your package does what I want. Please help.

**Comment by vedantjk** [comment]
So I found that I needed to use npm-init and did the following steps. Still giving error.
PS C:\Users\Vedant\Downloads\ServerForOTP-main> npm init
This utility will walk you through creating a package.json file.
It only covers the most common items, and tries to guess sensible defaults.

See `npm help init` for definitive documentation on these fields
and exactly what they do.

Use `npm install <pkg>` afterwards to install a package and
save it as a dependency in the package.json file.

Press ^C at any time to quit.
package name: (serverforotp-main) emailauth
version: (1.0.0)
description:
entry point: (index.js)
test command:
git repository:
keywords:
author:
license: (ISC)
About to write to C:\Users\Vedant\Downloads\ServerForOTP-main\package.json:

{
  "name": "emailauth",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}


Is this OK? (yes) yes
PS C:\Users\Vedant\Downloads\ServerForOTP-main> (npm i) -and (npm link)
npm WARN emailauth@1.0.0 No description
npm WARN emailauth@1.0.0 No repository field.

npm WARN emailauth@1.0.0 No description
npm WARN emailauth@1.0.0 No repository field.

True
PS C:\Users\Vedant\Downloads\ServerForOTP-main> email-auth-node
email-auth-node : The term 'email-auth-node' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try 
again.
At line:1 char:1
+ email-auth-node
+ ~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (email-auth-node:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

**Reply by saran-surya** [reply]
Can you kindly use Command prompt ?

**Reply by vedantjk** [reply]
I used the command prompt and now this:-
C:\Users\Vedant\Downloads\ServerForOTP-main>npm i && npm link
npm WARN emailauth@1.0.0 No description
npm WARN emailauth@1.0.0 No repository field.

up to date in 0.537s
found 0 vulnerabilities

npm WARN emailauth@1.0.0 No description
npm WARN emailauth@1.0.0 No repository field.

up to date in 0.432s
found 0 vulnerabilities

C:\Users\Vedant\AppData\Roaming\npm\node_modules\emailauth -> C:\Users\Vedant\Downloads\ServerForOTP-main

C:\Users\Vedant\Downloads\ServerForOTP-main>email-auth-node --version
'email-auth-node' is not recognized as an internal or external command,
operable program or batch file.

**Reply by vedantjk** [reply]
> Can you kindly use Command prompt ?

Can you tell me what to do now? I'm stuck and I am using this for an app.

**Reply by saran-surya** [reply]
use the command ```npx email-auth-node``` in command prompt

**Reply by vedantjk** [reply]
C:\Users\Vedant\Downloads\ServerForOTP-main>npm i
npm WARN emailauth@1.0.0 No description
npm WARN emailauth@1.0.0 No repository field.

up to date in 0.464s
found 0 vulnerabilities


C:\Users\Vedant\Downloads\ServerForOTP-main>npm link
npm WARN emailauth@1.0.0 No description
npm WARN emailauth@1.0.0 No repository field.

up to date in 0.466s
found 0 vulnerabilities

C:\Users\Vedant\AppData\Roaming\npm\node_modules\emailauth -> C:\Users\Vedant\Downloads\ServerForOTP-main

C:\Users\Vedant\Downloads\ServerForOTP-main>npx email-auth-node
npx: installed 62 in 3.46s
----------------- Package Help -----------------

Options available with 'email-auth-node' :
Options :
         --verify : This will verify the index.html file for the mandatory fields.
         >> usage >> npx email-auth-node --verify

         --generate : This will create a command line interface for generating the server config
         >> usage >> npx-email-auth-node --generate


email-auth-node >> New features are coming soon...

----------------- Package Help -----------------

C:\Users\Vedant\Downloads\ServerForOTP-main>email-auth-node
'email-auth-node' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\Vedant\Downloads\ServerForOTP-main>
C:\Users\Vedant\Downloads\ServerForOTP-main>email-auth-node --verify
'email-auth-node' is not recognized as an internal or external command,
operable program or batch file.

It still is not working at all.

**Reply by saran-surya** [reply]
Try using ```npx email-auth-node```

**Comment by saran-surya** [comment, marked as answer]
Kindly use  ```npx email-auth-node --version``` and for all the other
commands of email-auth-node you should prefix it with npx.

**Reply by vedantjk** [reply]
Ya it worked. Thanks man.

**Comment by saran-surya** [comment]
You also get the usage code in the package help directly, please take time
and read them too.
